pkg/controllers/repositoryserver/repositoryserver_controller_test.go

```diff
@@ -149,6 +157,186 @@ func (s *RepoServerControllerSuite) CreateGCPStorageCredentialsSecret(data map[s
 	return testutil.CreateSecret(s.kubeCli, "test-repository-server-storage-creds-", s.repoServerControllerNamespace, v1.SecretType(secrets.GCPSecretType), data)
 }
 
+func (s *RepoServerControllerSuite) TestRepositoryServerImmutability(c *C) {
+	// Create a repository server CR.
+	repoServerCR := testutil.GetTestKopiaRepositoryServerCR(s.repoServerControllerNamespace)
+	setRepositoryServerSecretsInCR(&s.repoServerSecrets, repoServerCR)
+	repoServerCRCreated, err := s.crCli.RepositoryServers(s.repoServerControllerNamespace).Create(context.Background(), repoServerCR, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+	// Update the repository server CR's Immutable field.
+	repoServerCRCreated.Spec.Repository.RootPath = "/updated-test-path/"
+	_, err = s.crCli.RepositoryServers(s.repoServerControllerNamespace).Update(context.Background(), repoServerCRCreated, metav1.UpdateOptions{})
+	// Expect an error.
+	c.Assert(err, NotNil)
+	// Delete the repository server CR.
+	err = s.crCli.RepositoryServers(s.repoServerControllerNamespace).Delete(context.Background(), repoServerCRCreated.Name, metav1.DeleteOptions{})
+	c.Assert(err, IsNil)
+}
+
+func (s *RepoServerControllerSuite) TestRepositoryServerStatusIsServerReady(c *C) {
+	// Test if the repository server CR state is Ready
+	// with all the right configuration
+	repoServerCR := testutil.GetTestKopiaRepositoryServerCR(s.repoServerControllerNamespace)
+	setRepositoryServerSecretsInCR(&s.repoServerSecrets, repoServerCR)
+	repoServerCRCreated, err := s.crCli.RepositoryServers(s.repoServerControllerNamespace).Create(context.Background(), repoServerCR, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+	err = s.waitForRepoServerInfoUpdateInCR(repoServerCRCreated.Name)
+	c.Assert(err, IsNil)
+	err = testutil.CreateTestKopiaRepository(s.kubeCli, repoServerCRCreated, testutil.GetDefaultS3CompliantStorageLocation())
+	c.Assert(err, IsNil)
+	err = s.waitOnRepositoryServerState(c, repoServerCRCreated)
+	c.Assert(err, IsNil)
+}
+
+func (s *RepoServerControllerSuite) TestRepositoryServerCRStateWithoutSecrets(c *C) {
+	// Test if server is stopped
+	// when no storage secrets are set
```

Comment:
Let's move comment above the test method and I see that we mention stopped in comment and we are waiting for failed in the code.

Reply:
Comment needs to be move for every test I think.